lib/node_modules/@stdlib/ndarray/array/docs/types/index.d.ts

```diff
@@ -164,7 +164,7 @@ interface ExtendedOptions extends Options {
 * var v = arr.get( 0 );
 * // returns [ 1, 2 ]
 */
-declare function array<T = unknown>( options: OptionsWithShape | OptionsWithBuffer ): typedndarray<T>;
+declare function array<T = unknown>( options: OptionsWithShape<T> | OptionsWithBuffer<T> ): typedndarray<typeof options['dtype'] extends undefined ? T : typeof options['dtype']>;
```

Comment:
I am not sure I follow how the return type is supposed to work. Can you provide examples of what you believe to be expected behavior?

Reply:
You should also add corresponding tests to `docs/types/test.ts` to confirm expected behavior. All our TypeScript declarations should have corresponding unit tests.